packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -355,85 +355,115 @@ IOSink? _openSink(String? output) {
   return sink;
 }
 
-/// A generator that will write code to a sink based on the contents of [PigeonOptions].
-abstract class Generator {
-  /// Returns an [IOSink] instance to be written to if the [Generator] should
-  /// generate.  If it returns `null`, the [Generator] will be skipped.
-  IOSink? shouldGenerate(PigeonOptions options);
+/// An adapter that will call a generator to write code to a sink
+/// based on the contents of [PigeonOptions].
+abstract class GeneratorAdapter {
+  /// Constructor for [GeneratorAdapter]
+  GeneratorAdapter(this.fileTypeList);
+
+  /// A list of [FileType]
```

Comment:
This is just a restatement of the code, so isn't a useful comment on its own. Let's combine it with the substance below to make the comment just:

```
/// A list of file types the generator should create.
```